### DISCUSSION TITLE
What are the expected units for cre command?

### DISCUSSION THREAD

**Discussion by MoraRubio** [TUDelft-CNS-ATM/bluesky#474] [Q&A] [answered]
Hi all, 

Looking at the wiki ([bluesky/wiki/Altitude](https://github.com/TUDelft-CNS-ATM/bluesky/wiki/Altitude)) it says that "The standard unit of entered altitudes is feet", but looking at the source code on bluesky/traffic/traffic.py, it says that `traffic.alt` is stored in meters (line 96), and then on the `cre` method, the acalt parameter is stored directly to `traffic.alt` (line 242), which leads to me conclude that acalt should be in meters.

What I understand then is that when using the CRE command (e.g. `CRE ac b737 52.2199 -0.4609 0 900 10`) the altitude should be in feet and speed in knots, but when using `traffic.cre` method the altitude should be in meters and speed in meters per second.

Could you please confirm or correct me if I'm seeing it wrong.

Thanks.

Alejandro

**Comment by jooste** [comment]
Hi @MoraRubio,

Internally, bluesky uses metric units almost exclusively (only some angles are in degrees). Nevertheless, most stack arguments default to imperial and nautical units, because these are the standard in aviation. Stack converts these automatically to metric units before passing them on to the respective function. Note that explicit unit specification when calling stack functions is on the TODO list. This should also make things clearer.

**Reply by MoraRubio** [reply]
Thanks for the prompt response, so this means that when using `traffic.cre` directly from a new Python plug-in I should use metric units?

**Reply by jooste** [reply, marked as answer]
Yes indeed. When you don't call the function via the stack use metric!
